<?xml version="1.0"?>
<robot name="motor_damiao" xmlns:xacro="http://www.ros.org/wiki/xacro">

    <ros2_control name="motor_damiao" type="system">

        <hardware>
            <plugin>motor_damiao_hardware/MotorDamiaoRobot</plugin>
            <!-- <plugin>gazebo_ros2_control/GazeboSystem</plugin> -->

            <param name="joint_states_publish_timeout">1.0</param>
            <param name="feedback_timeout">1.0</param>
            <param name="can0_baud_rate">1000000</param>
            <!-- <param name="can1_baud_rate">1000000</param> -->
        </hardware>

        <joint name="left_wheel_joint">
            <command_interface name="position" >
                <param name="min">-1.57</param>
                <param name="max">1.57</param>
            </command_interface>
            <command_interface name="velocity">
                <param name="min">-1.0</param>
                <param name="max">1.0</param>
            </command_interface>
            <command_interface name="effort">
                <param name="min">-5.0</param>
                <param name="max">5.0</param>
            </command_interface>
            <state_interface name="position" />
            <state_interface name="velocity" />
            <state_interface name="effort" />

            <param name="can_name">can0</param>
            <param name="can_id">0x01</param>
            <param name="feedback_id">0x11</param>
            <param name="position_max">12.5</param>
            <param name="velocity_max">45.0</param>
            <param name="kp">5.0</param>
            <param name="kd">1.0</param>
            <param name="torque_max">10.0</param>
            <param name="reverse">false</param>
            <param name="enable_position_expansion">true</param>
        </joint>
        <joint name="right_wheel_joint">
            <command_interface name="position" >
                <param name="min">-1.57</param>
                <param name="max">1.57</param>
            </command_interface>
            <command_interface name="velocity">
                <param name="min">-1.0</param>
                <param name="max">1.0</param>
            </command_interface>
            <command_interface name="effort">
                <param name="min">-5.0</param>
                <param name="max">5.0</param>
            </command_interface>
            <state_interface name="position" />
            <state_interface name="velocity" />
            <state_interface name="effort" />

            <param name="can_name">can0</param>
            <param name="can_id">0x02</param>
            <param name="feedback_id">0x12</param>
            <param name="position_max">12.5</param>
            <param name="velocity_max">45.0</param>
            <param name="kp">5.0</param>
            <param name="kd">1.0</param>
            <param name="torque_max">10.0</param>
            <param name="reverse">true</param>
            <param name="enable_position_expansion">true</param>
        </joint>
    </ros2_control>

</robot>